File: Java/Medium/WrongAnswer/number-of-ways-to-split-array_0.java
class Solution {
    public int waysToSplitArray(int[] nums) {
        
        long[] sum = new long[nums.length + 1];
        for (int i = 0; i < nums.length; i++) {
            sum[i + 1] = sum[i] + nums[i];
        }
        int ans = 0;
        for (int i = 0; i < nums.length - 2; i++) {
            long s = sum[i + 1];
            int l = i + 1, r = nums.length - 1;
            while (l < r) {
                int mid = (l + r) / 2;
                if (sum[mid + 1] - s >= s) {
                    r = mid;
                } else {
                    l = mid + 1;
                }
            }
            int left = l;
            l = i + 1;
            r = nums.length - 1;
            while (l < r) {
                int mid = (l + r + 1) / 2;
                if (sum[mid + 1] - s > sum[nums.length] - sum[mid + 1]) {
                    r = mid - 1;
                } else {
                    l = mid;
                }
            }
            int right = r;
            if (left <= right) {
                ans = (ans + right - left + 1) % 1000000007;
            }
        }
        return ans;
    }
}

File: C++/Medium/WrongAnswer/number-of-ways-to-split-array_0.cpp
class Solution {
public:
    int waysToSplitArray(vector<int>& nums) {
        
        int n = nums.size();
        vector<int> sum(n, 0);
        sum[0] = nums[0];
        for (int i = 1; i < n; ++i) {
            sum[i] = sum[i-1] + nums[i];
        }
        int ans = 0;
        for (int i = 0; i < n - 2; ++i) {
            int left = sum[i];
            int right = sum[n-1] - left;
            int mid = sum[n-1] - left;
            int low = i + 1;
            int high = n - 2;
            while (low <= high) {
                int mid = low + (high - low) / 2;
                int midLeft = sum[mid] - left;
                int midRight = right - midLeft;
                if (midLeft > midRight) {
                    high = mid - 1;
                } else {
                    low = mid + 1;
                }
            }
            int p = high;
            low = i + 1;
            high = n - 2;
            while (low <= high) {
                int mid = low + (high - low) / 2;
                int midLeft = sum[mid] - left;
                int midRight = right - midLeft;
                if (midLeft < midRight) {
                    low = mid + 1;
                } else {
                    high = mid - 1;
                }
            }
            int q = low;
            if (p >= q) {
                ans += p - q + 1;
                ans %= 1000000007;
            }
        }
        return ans;
    }
};